sonar-scanner-engine/src/main/java/org/sonar/scanner/genericcoverage/GenericTestExecutionSensor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.scanner.genericcoverage;
+
+import com.google.common.collect.ImmutableList;
+import java.io.File;
+import java.util.stream.Collectors;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.config.PropertyDefinition;
+import org.sonar.api.resources.Qualifiers;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.scanner.deprecated.test.TestPlanBuilder;
+
+import static org.sonar.api.CoreProperties.CATEGORY_CODE_COVERAGE;
+
+public class GenericTestExecutionSensor implements Sensor {
+
+  private static final Logger LOG = Loggers.get(GenericTestExecutionSensor.class);
+
+  private static final String REPORT_PATHS_PROPERTY_KEY = "sonar.testExecutionReportPaths";
+  /**
+   * @deprecated since 6.2
+   */
+  @Deprecated
+  private static final String OLD_UNIT_TEST_REPORT_PATHS_PROPERTY_KEY = "sonar.genericcoverage.unitTestReportPaths";
+
+  private final TestPlanBuilder testPlanBuilder;
+
+  public GenericTestExecutionSensor(TestPlanBuilder testPlanBuilder) {
+    this.testPlanBuilder = testPlanBuilder;
+  }
+
+  public static ImmutableList<PropertyDefinition> properties() {
+    return ImmutableList.of(
+
+      PropertyDefinition.builder(REPORT_PATHS_PROPERTY_KEY)
+        .name("Unit tests results report paths")
+        .description("List of comma-separated paths (absolute or relative) containing unit tests results report.")
+        .category(CATEGORY_CODE_COVERAGE)
+        .onQualifiers(Qualifiers.PROJECT)
+        .deprecatedKey(OLD_UNIT_TEST_REPORT_PATHS_PROPERTY_KEY)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "OLD_UNIT_TEST_REPORT_PATHS_PROPERTY_KEY"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)